FantasyUniverse/Services/Auth/FirebaseAuthService.swift

```diff
@@ -30,97 +36,159 @@ final class FirebaseAuthService: ObservableObject {
         state = .proccessing
         handle = Auth
             .auth()
-            .addStateDidChangeListener { (_, user) in
+            .addStateDidChangeListener { [weak self] (_, user) in
+                guard let self = self else { return }
                 if let user = user {
-                    print("Got user: \(user)")
                     
-                   let user = User(
-                        uid: user.uid,
-                        displayName: user.displayName,
-                        email: user.email,
-                        refreshToken: user.refreshToken)
-                    self.state = .success(with: user)
+                    self.fetchedUserInfoPublisher(uid: user.uid)
+                        .receive(on: RunLoop.main)
+                        .sink { res in
+                            switch res {
+                            case .failure(let err):
+                                    debugPrint(err.localizedDescription)
+                                    self.state = .failed(with: err)
+                                    self.signOut()
+                            default: break
+                            }
+                        } receiveValue: { userInfo in
+                            let user = User(
+                                uid: user.uid,
+                                email: user.email,
+                                displayName: user.displayName,
+                                refreshToken: user.refreshToken,
+                                profileImageURL: user.photoURL,
+                                firstName: userInfo.firstname,
+                                lastName: userInfo.lastname)
+                            self.state = .success(with: user)
+                            self.currentUserUID = user.uid
+                        }
+                        .store(in: &self.subscriptions)
                 } else {
                     self.state = .unauthenticated
+                    self.currentUserUID = nil
```

Comment:
Sounding like a broken record probably, but I'd rather have the ID as an associated value to the `authenticated` state and reduce possible combinations of states

Reply:
Right